### PR TITLE
Move release script to Python and run it with pixi

### LINKED ISSUE #987: Move release script to Python and run it with pixi
**What**
Currently, our release script is part of TeamCity and cannot easily be run locally: https://dpcbuild.deltares.nl/admin/editBuild.html?id=buildType:Ribasim_Ribasim_MakeGitHubRelease
The same is true for iMOD Coupler.
It would be good to move that script to Python and add a corresponding pixi task.

**Why**
As mentioned by @Manangka iMOD Python has similar needs, and as all three projects use pixi, it would be good to standardize on a common release process.

**How**

- [ ] Move [script](https://dpcbuild.deltares.nl/admin/editBuild.html?id=buildType:Ribasim_Ribasim_MakeGitHubRelease) to Python
- [ ] Add pixi task to run the python script
- [ ] Use pixi task on Teamcity for Ribasim and iMOD Coupler

### PR DESCRIPTION
**Pull request by Hofer-Julian**
Fixes #987

### CODE REVIEW

**Comment by Hofer-Julian** [comment]
@Manangka

File: pixi.toml [hunk 1]
```diff
@@ -161,6 +163,7 @@ typing-extensions = ">=4.6"
 xarray = "*"
 xmipy = "*"
 xugrid = "*"
+gh = "2.43.1.*"
```
Comment (visr) [inline pixi.toml:166]:
```suggestion
gh = "*"
```

**Review by visr — APPROVED** [review]

File: .teamcity/Ribasim/buildTypes/Ribasim_Ribasim_MakeGitHubRelease.xml [hunk 2]
```diff
@@ -13,27 +13,9 @@
...
-        --generate-notes \
-        ribasim_cli_linux.zip \
-        ribasim_cli_windows.zip \
-        ribasim_qgis.zip \
-        generated_testmodels.zip
-
-    echo "Release created successfully."
-
-else
-    echo "No tag is currently checked out."
```
Comment (visr) [inline .teamcity/Ribasim/buildTypes/Ribasim_Ribasim_MakeGitHubRelease.xml:35]: Don't you need to check for the "No tag is currently checked out." condition anymore?

Comment (Hofer-Julian) [reply .teamcity/Ribasim/buildTypes/Ribasim_Ribasim_MakeGitHubRelease.xml:35]: It's not needed. I tested it locally and git returns with an error code if there's no tag.

**Pull request merged**